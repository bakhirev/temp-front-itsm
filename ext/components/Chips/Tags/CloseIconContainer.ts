import styled, { css } from 'styled-components';

import { DEFAULT_THEME } from '../../common';
import { HEIGHT_BIG, HEIGHT_SMALL, MARGIN_LEFT_BIG, MARGIN_LEFT_SMALL } from '../common/constants';

import type { IChipTagProps } from './ChipTagItem';

export const CloseIconContainer = styled.div<IChipTagProps>`
  ${({ disabled, size, theme: { color } }) => css`
    display: flex;
    align-items: center;
    justify-content: center;
    height: ${size === 'big' ? HEIGHT_BIG : HEIGHT_SMALL};
    width: ${size === 'big' ? HEIGHT_BIG : HEIGHT_SMALL};
    border-radius: 50%;
    margin-left: ${size === 'big' ? MARGIN_LEFT_BIG : MARGIN_LEFT_SMALL};
    &:hover {
      background-color: ${disabled ? 'none' : color.opacity.blackHover};
      outline: none;
      cursor: ${!disabled ? 'pointer' : 'default'};
    }
    &:focus {
      outline: none;
      border: none;
      background-color: ${disabled ? 'none' : color.opacity.blackFocus};
    }
    &:active {
      outline: none;
      border: none;
      background-color: ${disabled ? 'none' : color.opacity.blackPressed};
    }
  `};
`;

CloseIconContainer.defaultProps = {
  theme: DEFAULT_THEME,
};
