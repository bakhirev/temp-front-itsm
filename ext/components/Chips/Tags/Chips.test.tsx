import React from 'react';
import { shallow } from 'enzyme';

import { CloseIconContainer } from './CloseIconContainer';

import { IItemChipTag, Tags } from './index';

const listData: IItemChipTag[] = [
  { id: '1', label: 'Суздаль' },
  { id: '2', label: 'Москва', disabled: true },
];

const listItemsDisabled: IItemChipTag[] = [
  { id: 'a', label: 'Суздаль', disabled: true },
  { id: 'b', label: 'Москва', disabled: true },
  { id: 'c', label: 'Тверь', disabled: true },
  { id: 'd', label: 'Самара', disabled: true },
];

jest.mock('../../common', () => ({
  DEFAULT_THEME: {},
  Z_INDEXES: {
    TOOLTIP: 777,
  },
}));

describe('Tags', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render component with items', () => {
    const wrapper = shallow(
      <Tags
        items={listData}
        size={'big'}
        onRemoveItem={(id) => listData.filter((d) => d.id !== id)}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render component with disabled items', () => {
    const wrapper = shallow(<Tags items={listItemsDisabled} size={'big'} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should call onClick when use click on component', () => {
    const chipsRequiredProps = {
      onRemoveItem: jest.fn(),
    };
    const wrapper = shallow(<Tags items={listData} size={'big'} {...chipsRequiredProps} />);
    wrapper.childAt(0).find(CloseIconContainer).simulate('click');
    const { onRemoveItem } = chipsRequiredProps;
    expect(onRemoveItem).toBeCalledTimes(1);
  });

  it('should call onClick when use click on disabled component', () => {
    const chipsRequiredProps = {
      onRemoveItem: jest.fn(),
    };
    const wrapper = shallow(
      <Tags items={listItemsDisabled} size={'big'} {...chipsRequiredProps} />
    );
    wrapper.childAt(0).find(CloseIconContainer).simulate('click');
    const { onRemoveItem } = chipsRequiredProps;
    expect(onRemoveItem).toBeCalledTimes(0);
  });
});
