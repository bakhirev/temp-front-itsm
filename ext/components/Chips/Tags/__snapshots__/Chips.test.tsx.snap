// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tags should render component with disabled items 1`] = `
<styled.div>
  <styled.div
    disabled={true}
    size="big"
    theme={Object {}}
    tooltip="Суздаль"
  >
    <styled.div
      size="big"
    >
      <Styled(styled.div)>
        Суздаль
      </Styled(styled.div)>
    </styled.div>
    <styled.div
      disabled={true}
      size="big"
      tabIndex={-1}
      theme={Object {}}
    >
      <styled.div
        height={20}
        width={20}
      />
    </styled.div>
  </styled.div>
  <styled.div />
  <styled.div
    disabled={true}
    size="big"
    theme={Object {}}
    tooltip="Москва"
  >
    <styled.div
      size="big"
    >
      <Styled(styled.div)>
        Москва
      </Styled(styled.div)>
    </styled.div>
    <styled.div
      disabled={true}
      size="big"
      tabIndex={-1}
      theme={Object {}}
    >
      <styled.div
        height={20}
        width={20}
      />
    </styled.div>
  </styled.div>
  <styled.div />
  <styled.div
    disabled={true}
    size="big"
    theme={Object {}}
    tooltip="Тверь"
  >
    <styled.div
      size="big"
    >
      <Styled(styled.div)>
        Тверь
      </Styled(styled.div)>
    </styled.div>
    <styled.div
      disabled={true}
      size="big"
      tabIndex={-1}
      theme={Object {}}
    >
      <styled.div
        height={20}
        width={20}
      />
    </styled.div>
  </styled.div>
  <styled.div />
  <styled.div
    disabled={true}
    size="big"
    theme={Object {}}
    tooltip="Самара"
  >
    <styled.div
      size="big"
    >
      <Styled(styled.div)>
        Самара
      </Styled(styled.div)>
    </styled.div>
    <styled.div
      disabled={true}
      size="big"
      tabIndex={-1}
      theme={Object {}}
    >
      <styled.div
        height={20}
        width={20}
      />
    </styled.div>
  </styled.div>
</styled.div>
`;

exports[`Tags should render component with items 1`] = `
<styled.div>
  <styled.div
    size="big"
    theme={Object {}}
    tooltip="Суздаль"
  >
    <styled.div
      size="big"
    >
      <Styled(styled.div)>
        Суздаль
      </Styled(styled.div)>
    </styled.div>
    <styled.div
      onClick={[Function]}
      size="big"
      tabIndex={1}
      theme={Object {}}
    >
      <styled.div
        height={20}
        width={20}
      />
    </styled.div>
  </styled.div>
  <styled.div />
  <styled.div
    disabled={true}
    size="big"
    theme={Object {}}
    tooltip="Москва"
  >
    <styled.div
      size="big"
    >
      <Styled(styled.div)>
        Москва
      </Styled(styled.div)>
    </styled.div>
    <styled.div
      disabled={true}
      size="big"
      tabIndex={-1}
      theme={Object {}}
    >
      <styled.div
        height={20}
        width={20}
      />
    </styled.div>
  </styled.div>
</styled.div>
`;
